.github/workflows/mend-cli-scan.yaml

```diff
@@ -49,45 +41,118 @@ jobs:
+          mend dep --no-color -s ${{ vars.MEND_PRODUCT_NAME }}//${{ vars.MEND_PROJECT_NAME }} -u > mend-sca-scan-result.txt
+
+          export MEND_SCA_SCAN_URL=$(grep -Eo '(http|https)://[^ ]+' mend-sca-scan-result.txt)
+          export MEND_SCA_SCAN_SUMMARY=$(grep -Eo 'Detected [0-9]+ vulnerabilities.*' mend-sca-scan-result.txt)
+          export MEND_SCA_CRITICAL_COUNT=$(grep -Eo '[0-9]+ Critical' mend-sca-scan-result.txt | grep -Eo '[0-9]+')
+          export MEND_SCA_HIGH_COUNT=$(grep -Eo '[0-9]+ High' mend-sca-scan-result.txt | grep -Eo '[0-9]+')
+          export MEND_SCA_MEDIUM_COUNT=$(grep -Eo '[0-9]+ Medium' mend-sca-scan-result.txt | grep -Eo '[0-9]+')
+          export MEND_SCA_LOW_COUNT=$(grep -Eo '[0-9]+ Low' mend-sca-scan-result.txt | grep -Eo '[0-9]+')
+
+          echo "MEND_SCA_SCAN_URL=$MEND_SCA_SCAN_URL" >> $GITHUB_ENV
+          echo "MEND_SCA_SCAN_SUMMARY=$MEND_SCA_SCAN_SUMMARY" >> $GITHUB_ENV
+          echo "MEND_SCA_CRITICAL_COUNT=$MEND_SCA_CRITICAL_COUNT" >> $GITHUB_ENV
+          echo "MEND_SCA_HIGH_COUNT=$MEND_SCA_HIGH_COUNT" >> $GITHUB_ENV
+          echo "MEND_SCA_MEDIUM_COUNT=$MEND_SCA_MEDIUM_COUNT" >> $GITHUB_ENV
+          echo "MEND_SCA_LOW_COUNT=$MEND_SCA_LOW_COUNT" >> $GITHUB_ENV
 
-      # Check for failures and set the outcome of the workflow
-      - name: Parse the result and set job status
+      # Check for failures in SCA scan and set the outcome of the workflow
+      - name: Fail if Critical or High SCA vulnerabilities are found
+        shell: bash
+        run: |
+          if [ "$MEND_SCA_CRITICAL_COUNT" -gt 0 ] || [ "$MEND_SCA_HIGH_COUNT" -gt 0 ]; then
+            echo "❌ SCA scan detected critical/high vulnerabilities."
+            exit 1
+          else
+            echo "✅ No critical/high SCA vulnerabilities."
+          fi
+
+      # Publish the Mend SCA scan result
+      - name: Mend SCA Scan Result
+        uses: LouisBrunner/checks-action@v1.6.1
         if: always()
+        with:
+          name: "Mend SCA Scan Result"
+          token: ${{ secrets.GITHUB_TOKEN }}
+          conclusion: ${{ job.status }}
+          output_text_description_file: mend-sca-scan-result.txt
+          output: |
+            {"title":"Mend SCA Scan Result", "summary":"${{ job.status }}"}
+
+      ########################################
+      #       Execute Mend SAST scan         #
+      ########################################
+      - name: Mend SAST scan
+        env:
+          MEND_EMAIL: ${{ secrets.MEND_EMAIL }}
+          MEND_USER_KEY: ${{ secrets.MEND_USER_KEY }}
+          MEND_URL: ${{ vars.MEND_SERVER_URL }}
+        shell: bash
+        run: |
+          mend code --report --filename ${{ vars.MEND_SAST_REPORT_NAME }} --formats json,pdf --non-interactive --scope ${{ vars.MEND_PRODUCT_NAME }}//${{ vars.MEND_PROJECT_NAME }} > mend-sast-scan-result.txt
```

Comment:
We run the mend SAST scan and produce 2 files - json and pdf + the report is published to Mend under ${{ vars.MEND_PRODUCT_NAME }}//${{ vars.MEND_PROJECT_NAME }}
The json file is used to parse the results. The pdf is attached to the pipeline run... (see below...)